csp/middleware.py

```diff
@@ -50,35 +48,48 @@ def process_response(self, request, response):
             return response
 
         # Check for ignored path prefix.
-        prefixes = getattr(settings, 'CSP_EXCLUDE_URL_PREFIXES', ())
+        # TODO: Legacy setting
+        prefixes = getattr(
+            settings,
+            'CSP_EXCLUDE_URL_PREFIXES',
+            defaults.EXCLUDE_URL_PREFIXES,
+        )
         if request.path_info.startswith(prefixes):
             return response
 
         # Check for debug view
-        status_code = response.status_code
-        exempted_debug_codes = (
-            http_client.INTERNAL_SERVER_ERROR,
-            http_client.NOT_FOUND,
-        )
-        if status_code in exempted_debug_codes and settings.DEBUG:
+        if response.status_code in EXEMPTED_DEBUG_CODES and settings.DEBUG:
             return response
 
-        header = 'Content-Security-Policy'
-        if getattr(settings, 'CSP_REPORT_ONLY', False):
-            header += '-Report-Only'
-
-        if header in response:
+        existing_headers = {
+            header for header in HTTP_HEADERS if header in response
+        }
+        if len(existing_headers) == len(HTTP_HEADERS):
             # Don't overwrite existing headers.
             return response
 
-        response[header] = self.build_policy(request, response)
+        # These won't be ordered properly on Cpython 3.5 or below
```

Comment:
Remove October 2020 comments and OrderedDict (not supporting python 3.5 anymore?)

Reply:
Done